pygount/analysis.py

```diff
@@ -521,6 +551,47 @@ def name_regexps_to_skip(self):
     def name_regexps_to_skip(self, regexps_or_pattern_text):
         self._name_regexp_to_skip = pygount.common.regexes_from(regexps_or_pattern_text, self.name_regexps_to_skip)
 
+    @staticmethod
+    def is_valid_git_link(git_link: str) -> bool:
+        # Regex to check valid  GIT Repository
+        # from https://stackoverflow.com/questions/2514859/regular-expression-for-git-repository
+        git_regex = re.compile(r"((git|ssh|http(s)?)|(git@[\w\.-]+))(:(\/\/)?)([\w\.@\:\/\-~]+)(\.git)(\/)?")
+
+        if git_regex.match(git_link):
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def repo_arguments(git_link: str):
+        """
+        Returns the git link that ends with .git and branch name if it specified at the end with @branch
+        :param git_link: input of the git link that ends with .git or .git@branch
+        """
+        # example: master, feature-branch
+        # at this state it's set to '@master' or ''
+        git_branch = git_link.split(".git")[-1]
+
+        # example: https://github.com/roskakori/pygount.git
+        git_link = git_link.split(".git")[0] + ".git"
+
+        # set the argument for the clone command if any branch is specified after `.git/`
+        if git_branch != "":
+            # set from '@master' to 'master'
+            git_branch = "--branch " + git_branch.split("@")[1]
+
+        return git_link, git_branch
+
+    def _set_temp_path_from_git_link(self, source_patterns):
+        if source_patterns != [] and self.is_valid_git_link(source_patterns[0]):
+            self.is_git_link = True
+            temp_folder = mkdtemp()
+            git_link, git_branch = self.repo_arguments(source_patterns[0])
+
+            git.Repo.clone_from(git_link, temp_folder, multi_options=["--depth 1", git_branch])
```

Comment:
AFAIR github doesn't allow to use `git archive --remote`: https://github.com/isaacs/github/issues/554.
Another solution would be to manually download from github using archives.

e.g.:
https://github.com/roskakori/pygount/releases/tag/v1.5.1
https://github.com/roskakori/pygount/archive/master.zip

But they have a different syntax for branches and tags/releases and it would be hard to differentiate between the two

Reply:
Agreed, we need to use `git clone`.